src/org/typo3/chefci/v2/stages/Publish.groovy

```diff
@@ -21,6 +21,8 @@ class Publish extends AbstractStage {
      */
     protected publish() {
 
+        (new Slack(script)).notifyVersionBump()
```

Comment:
Hm - depends on what you are trying to achieve. If you want to have "generic" notifications and a sink (or many sinks) that handles such a notification, you could have a event that gets "thrown" and can register an `EventHandler` with a concrete implementation `SlackEventHandler` that processes such events.

But to keep it simple, I'd just go with the `Slack` class, but instead of instantiating it each time you call a method on it, instantiate it once (e.g. when the stage is created and registered in the Pipeline). Then you can have the instantiation once for the Pipeline (it should stay the same for all stages in the Pipeline) and have it injected to the Stage, once it is registered at the Pipeline...